packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -0,0 +1,829 @@
+
+        operationId = keccak256(
+            abi.encode(
+                block.chainid,
+                destinationChainId,
+                asset,
+                amount,
+                recipient,
+                additionalData,
+                currentNonce,
+                operationType
+            )
+        );
+
+        // Set initial status to QUEUED
+        operationStatuses[operationId] = BridgeTypes.OperationStatus.QUEUED;
+
+        return operationId;
+    }
+
+    /**
+     * @dev Internal function to apply fee buffer for cross-chain operation volatility
+     * @param baseFee The base fee amount to buffer
+     * @return bufferedFee The fee with 1% buffer applied
+     */
+    function _applyFeeBuffer(
+        uint256 baseFee
+    ) internal pure returns (uint256 bufferedFee) {
+        // Add 1% buffer to account for fee volatility
+        return (baseFee * 101) / 100;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                           BRIDGE QUEUE OPERATIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @inheritdoc IBridgeRouter
+     */
+    function executeTransferAssets(
+        BridgeTypes.ExecuteTransferParams calldata params
+    )
+        external
+        payable
+        onlyBridgeQueue
+        whenNotPaused
+        nonReentrant
+        returns (bytes32 operationId)
+    {
+        _validateTransferParams(params);
+
+        // Get required base fee and specified adapter (no multiplier)
+        (uint256 requiredBaseFee, , address specifiedAdapter) = _quote(
+            params.destinationChainId,
+            params.asset,
+            params.amount,
+            params.options,
+            BridgeTypes.OperationType.TRANSFER_ASSET
+        );
+
+        // Apply fee buffer to account for fee volatility
+        uint256 bufferedFee = _applyFeeBuffer(requiredBaseFee);
+
+        // Validate fee provided by BridgeQueue against buffered fee
+        _validateFee(msg.value, bufferedFee);
+
+        _validateAdapterSupportsOperation(
+            specifiedAdapter,
+            BridgeTypes.OperationType.TRANSFER_ASSET
+        );
+
+        // Pull tokens from BridgeQueue to Router first
+        IERC20(params.asset).safeTransferFrom(
+            bridgeQueue, // BridgeQueue approved us
+            address(this), // Transfer to Router
+            params.amount
+        );
+
+        // Now approve the adapter to spend Router's tokens
+        IERC20(params.asset).approve(specifiedAdapter, 0);
+        IERC20(params.asset).approve(specifiedAdapter, params.amount);
+
+        // Notify originator that assets are now officially in-flight
+        // Attempt to call updateInflightAssets if the originator supports it
+        if (params.originator.code.length > 0) {
+            try
+                IERC165(params.originator).supportsInterface(
+                    type(IInflightAssetTracking).interfaceId
```

Comment:
there is a mixture of explicit (onchainregistry) and pretty generic (code size > 0 + interface compatibility) across the codebase, this should be unified, unless there's a specific reason for that.

In this case we already check if the caller is bridgeQueue - so that should be verfied once